src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -191,6 +191,46 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                             writer.Line($");");
                         }
                         break;
+                    case MultipartRequestBody multipartRequestBody:
+                        var multipartContent = new CodeWriterDeclaration("content");
+                        writer.Line($"var {multipartContent:D} = new {typeof(MultipartFormDataContent)}();");
+
+                        foreach (var bodyParameter in multipartRequestBody.RequestBodyParts)
+                        {
+                            switch (bodyParameter.Content)
+                            {
+                                case BinaryRequestBody binaryBody:
+                                    using (WriteValueNullCheck(writer, binaryBody.Value))
+                                    {
+                                        writer.Append($"{multipartContent}.Add({typeof(RequestContent)}.Create(");
+                                        WriteConstantOrParameter(writer, binaryBody.Value);
+                                        writer.Append($"), \"{bodyParameter.Name}");
+                                        writer.Line($"\", null);");
+                                    }
+                                    break;
+                                case TextRequestBody textBody:
+                                    using (WriteValueNullCheck(writer, textBody.Value))
+                                    {
+                                        writer.Append($"{multipartContent}.Add(new {typeof(StringRequestContent)}(");
+                                        WriteConstantOrParameter(writer, textBody.Value);
+                                        writer.Append($"), \"{bodyParameter.Name}");
+                                        writer.Line($"\", null);");
+                                    }
+                                    break;
```

Comment:
I don't know if it can be reused easily without significant refactoring because the statements within the [cases of multipart](https://github.com/Azure/autorest.csharp/blob/b9ec2da1dad2b4aa158cdc351923fd7f5193b964/src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs#L202-L219) are different than [this](https://github.com/Azure/autorest.csharp/blob/b9ec2da1dad2b4aa158cdc351923fd7f5193b964/src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs#L202-L219).